membership-attribute-service/app/controllers/AttributeController.scala

```diff
@@ -107,114 +105,11 @@ class AttributeController extends Controller with LoggingWithLogstashFields {
-            maybeDynamoAttributes map { dynamoAttributes =>
-              zuoraAttributes.copy(
-                AdFree = dynamoAttributes.AdFree, //fetched from Dynamo in the Zuora lookup anyway (dynamo is the source of truth)
-                Wallet = dynamoAttributes.Wallet, //can't be found based on Zuora lookups, and not currently used
-                MembershipNumber = dynamoAttributes.MembershipNumber, //I don't think membership number is needed and it comes from Salesforce
-                MembershipJoinDate = dynamoAttributes.MembershipJoinDate.flatMap(_ => zuoraAttributes.MembershipJoinDate), //only compare if dynamo has value
-                DigitalSubscriptionExpiryDate = None
-              )
-            }
-          }
-          if (zuoraAttributesWithIgnoredFields != maybeDynamoAttributes)
-            log.info(s"We looked up attributes via Zuora for $identityId and Zuora and Dynamo disagreed." +
-              s" Zuora attributes: $maybeZuoraAttributes. Dynamo attributes: $maybeDynamoAttributes.")
-        }
-      }
-    }
-
-    val attributesDisjunction = for {
-      accounts <- EitherT[Future, String, QueryResponse](withTimer(s"ZuoraAccountIdsFromIdentityId", zuoraAccountsQuery(identityId)))
-      accountIds = queryToAccountIds(accounts)
-      subscriptions <- EitherT[Future, String, List[Subscription[AnyPlan]]](
-        if(accountIds.nonEmpty) withTimer(s"ZuoraGetSubscriptions", getSubscriptions(accountIds))
-        else Future.successful(\/.right {
-          log.info(s"User with identityId $identityId has no accountIds and thus no subscriptions.")
-          Nil
-        })
-      )
-    } yield {
-      AttributesMaker.attributes(identityId, subscriptions, LocalDate.now())
-    }
-
-    val attributes = attributesDisjunction.run.map {
-      _.leftMap { errorMsg =>
-        log.error(s"Tried to get Attributes for $identityId but failed with $errorMsg")
-        errorMsg
-      }.fold(_ => None, identity)
-    }
-
-    val attributesFromDynamo: Future[Option[Attributes]] = attributeService.get(identityId)
-
-    compareThenLogAttributes(attributesFromDynamo, attributes)
-
-    val adFreeFlagFromDynamo = attributesFromDynamo map (_.flatMap(_.AdFree))
-
-    attributes flatMap { maybeAttributes =>
-      adFreeFlagFromDynamo map { adFree =>
-        maybeAttributes map {_.copy(AdFree = adFree)}
-      }
-    }
-  }
-
   private def zuoraLookup(endpointDescription: String) =
     backendAction.async { implicit request =>
       authenticationService.userId(request) match {
         case Some(identityId) =>
-          attributesFromZuora(identityId, request.touchpoint.patientZuoraRestService, request.touchpoint.subService, request.touchpoint.attrService).map {
+          getAttributes(identityId, request.touchpoint.zuoraRestService.getAccounts, accountId => reads => request.touchpoint.subService.subscriptionsForAccountId[AnyPlan](accountId)(reads), request.touchpoint.attrService.get).map {
```

Comment:
This looks a lot like the getAttributes call in line 58, could we extract this to a shared def that takes identity id and request as parameter ? It might clean up this long line as well

Reply:
I need to just totally remove that endpoint and that function but I couldn't decide if this PR was when I wanted to do it.